docs/source/API/core/MultiGPUSupport.rst

```diff
@@ -0,0 +1,153 @@
+.. role:: cppkokkos(code)
+    :language: cppkokkos
+
+Multi-GPU Support
+=================
+
+Kokkos has support for launching kernels on multiple GPU devices within a single node from host on a single MPI rank for the CUDA, HIP, and SYCL backends.
+
+Using this feature requires knowledge of backend specific API calls for creating non-default execution space instances. Once the execution space has been
+created, it can be used to create execution policies, allocate views, etc. all on the device chosen by the user.
+
+Constructing Execution Spaces
+-----------------------------
+
+CUDA
+~~~~
+
+For CUDA backend, the user creates a ``cudaStream_t`` object and passes it to the ``Kokkos::Cuda`` constructor.
+
+.. note:: The user is expected to manage the lifetime of the stream. Only after the execution space is destroyed
+          can the user safely destroy the stream.
+
+
+.. code-block:: cpp
+
+    // Query number of devices available
+    int n_devices;
+    cudaGetDeviceCount(&n_devices);
+
+    // Choose some 0 <= N < n_devices
+    int N = ...;
+
+    // Create stream on device N
+    cudaStream_t stream;
+    cudaSetDevice(N);
+    cudaStreamCreate(&stream);
+
+    // Scope execution space to ensure stream
+    // is destroyed *after* execution space
+    {
+      // Create execution space
+      Kokkos::Cuda exec_space(stream);
+
+      // Use execution space
+      /* ... */
+    }
+
+    // Destroying stream after use
+    cudaSetDevice(N);
+    cudaStreamDestroy(stream);
+
+HIP
+~~~
+
+For the HIP backend, like with CUDA, the user creates a ``hipStream_t`` object and passes it to the ``Kokkos::HIP`` constructor.
+
+.. note:: The user is expected to manage the lifetime of the stream. Only after the execution space is destroyed
+          can the user safely destroy the stream.
+
+.. warning:: Multi-GPU only supported for ROCm 5.6 and later. Because of the lack of HIP API functions for querying a
+             stream's device before ROCm 5.6, constructing a ``Kokkos::HIP`` instance on a non-default device isn't
+             supported, and problems will likely arise downstream.
+
+
+.. code-block:: cpp
+
+    // Query number of devices available
+    int n_devices;
+    hipGetDeviceCount(&n_devices);
+
+    // Choose some 0 <= N < n_devices
+    int N = ...;
+
+    // Create stream on device N
+    hipStream_t stream;
+    hipSetDevice(N);
+    hipStreamCreate(&stream);
+
+    // Scope execution space to ensure stream
+    // is destroyed *after* execution space
+    {
+      // Create execution space
+      Kokkos::HIP exec_space(stream);
+
+      // Use execution space
+      /* ... */
+    }
+
+    // Destroying stream after use
+    hipSetDevice(N);
+    hipStreamDestroy(stream);
+
+SYCL
+~~~~
+
+For the SYCL backend, the user creates a ``sycl::queue`` object and passes it to the ``Kokkos::SYCL`` constructor.
+
```

Comment:
We should also add the note here ... or move it to a different place for all of them.

Reply:
I thought `sycl::queue` did not need to be managed manually? I'm a bit out of my depth here though, maybe @masterleinad can back me up or refute..

Reply:
Yes, `sycl::queue` behaves like a `std::shared_ptr`.